commons/zenoh-codec/src/network/declare.rs

```diff
@@ -104,9 +108,13 @@ where
 
         // Header
         let mut header = id::DECLARE;
-        if x.interest_id.is_some() {
-            header |= declare::flag::I;
-        }
+        header |= match mode {
```

Comment:
Just a suggestion: Could having both `Request` and `RequestContinuous` have the heigh weight bit to 1 allow to easily distinguish Requests in some cases ?